job-dsl-core/src/test/groovy/javaposse/jobdsl/dsl/helpers/triggers/TriggerContextSpec.groovy

```diff
@@ -436,6 +436,65 @@ class TriggerContextSpec extends Specification {
         1 * mockJobManagement.requireMinimumPluginVersion('ghprb', '1.15-0')
     }
 
+    def 'call pull request trigger with extensions node and all args'() {
+        when:
+        context.pullRequest {
+            admins(['test'])
+            userWhitelist(['test'])
+            orgWhitelist(['test'])
+            cron('*/5 * * * *')
+            triggerPhrase('ok to test')
+            onlyTriggerPhrase(true)
+            useGitHubHooks(true)
+            allowMembersOfWhitelistedOrgsAsAdmin(true)
+            permitAll(true)
+            autoCloseFailedPullRequests(true)
+            commentFilePath('myCommentFile')
+
+            commitStatusContext('Deploy to staging site')
+            triggeredStatus('deploy triggered')
+            startedStatus('deploy started')
+
+            buildResultSuccessMessage('All is well')
+            buildResultFailureMessage('Something has gone wrong')
+        }
+
+        then:
+        with(context.triggerNodes[0]) {
+            name() == 'org.jenkinsci.plugins.ghprb.GhprbTrigger'
+            children().size() == 13
+            adminlist[0].value() == 'test'
+            whitelist[0].value() == 'test'
+            orgslist[0].value() == 'test'
+            cron[0].value() == '*/5 * * * *'
+            spec[0].value() == '*/5 * * * *'
+            triggerPhrase[0].value() == 'ok to test'
+            onlyTriggerPhrase[0].value() == true
+            useGitHubHooks[0].value() == true
+            allowMembersOfWhitelistedOrgsAsAdmin[0].value() == true
+            permitAll[0].value() == true
+            autoCloseFailedPullRequests[0].value() == true
+            commentFilePath[0].value() == 'myCommentFile'
+        }
+        with(context.triggerNodes[0].children()[12]) {
```

Comment:
Use nested with blocks to test sub-elements, see https://github.com/jenkinsci/job-dsl-plugin/blob/job-dsl-1.36/job-dsl-core/src/test/groovy/javaposse/jobdsl/dsl/helpers/publisher/PublisherContextSpec.groovy#L3339 for an example.